types/checksum.go

```diff
@@ -10,6 +10,10 @@ import (
 // The length of a checksum must always be ChecksumLen.
 type Checksum []byte
 
+func (cs Checksum) EncodeHex() string {
+	return hex.EncodeToString(cs)
+}
```

Comment:
We could implement this standard interface instead: https://pkg.go.dev/fmt#Stringer

The string representation of `Checksum` should always be lower case hex

Reply:
Implemented the `Stringer` interface for the type instead